spring-xd-messagebus-spi/src/main/java/org/springframework/xd/dirt/integration/bus/MessageBusSupport.java

```diff
@@ -1039,27 +1035,37 @@ public T lookupSharedChannel(String name) {
 	 */
 	abstract static class JavaClassMimeTypeConversion {
 
+		public static final MimeType APPLICATION_OCTET_STREAM_MIME_TYPE = MimeType.valueOf(APPLICATION_OCTET_STREAM_VALUE);
```

Comment:
There isn't a cache for invocations of `mimeTypeFromObject` that references APPLICATION_OCTET_STREAM_MIME_TYPE but there is a cache for invocations of `classNameFromMimeType.

In our perf tests we are relying on the fact that the object is an instance of byte[], so calls to MimeType.valueOf are bypassed.  It maybe relevant if the payload isn't of type byte[] or String since that is what leads to a call that does `MimeType.valueOf`.  

If we need this change, it should be part of another issue.